### PR TITLE
[Feature] Implement form `reset`

### LINKED ISSUE #62: [Feature Request]: Clear, Reset form state, value, error.
### Description

Easy way to reset/clear the form once submitted.

```
onSubmit={async (values, form) => {
  await postCreateMutation.mutateAsync({ title: values.title, content: valu.es.content });
  await postAllQuery.refetch();
  form.reset(); // This does not exist!
}}
```

form.reset(); should clear all errors or state of the form and fields in general.

### PR DESCRIPTION
**Pull request by ShooTeX**
closes #62

### CODE REVIEW

**Review by crutchcorn — CHANGES_REQUESTED** [review]
Looks great overall! Some minor notes

File: lib/form/form.spec.tsx [hunk 1]
```diff
@@ -1056,3 +1056,95 @@ test("Form's memoChild should prevent re-renders", async () => {
...
+        </button>
+        <p>
+          isNotDirty
+        </p>
+        <p>
+          isNotTouched
+        </p>
+      </div>
+      `)
+  );
```
[the remark this answers is not in the document]
Comment (ShooTeX) [reply lib/form/form.spec.tsx:1149]: will do!

File: lib/form/form.tsx [hunk 2]
```diff
@@ -169,6 +169,21 @@ function FormComp<T extends Record<string, any> = Record<string, any>>(
...
+
+      if (Array.isArray(field.value)) {
+        (field as FieldArrayInstance).setValues(value || []);
+      } else {
+        (field as FieldInstance).setValue(value || "");
+      }
+    });
+    _setIsDirty(false);
+    _setIsTouched(false);
+    _setErrors(null);
```
[the remark this answers is not in the document]
Comment (ShooTeX) [reply lib/form/form.tsx:185]: the initialValue of `_errors` is `null` and I don't see it set to `[]` or am I missing something?

**Comment by ShooTeX** [comment]
there is still a weird side effect where field arrays will reset to an empty field after submitting multiple times, can't really tell why this is happening

https://user-images.githubusercontent.com/10850738/226752199-ec440f1b-661d-47c3-b4d2-98d8a071c340.mov

**Comment by crutchcorn** [comment]
@ShooTeX this issue should be fixed now. I'll also fix the merge conflicts for ya :)

**Comment by crutchcorn** [comment]
@all-contributors please add @caxco93 for code, test, doc

**Comment by crutchcorn** [comment]
Oh whoops that's the wrong username 🤦

**Comment by crutchcorn** [comment]
@all-contributors please add @ShooTeX for code, test, doc

**Pull request merged**